iocore/net/QUICNetVConnection.cc

```diff
@@ -204,28 +203,40 @@ QUICNetVConnection::free(EThread *t)
 {
   QUICConDebug("Free connection");
 
-  this->_ctable->erase(this->_original_quic_connection_id, this);
-  this->_ctable->erase(this->_quic_connection_id, this);
-  for (unsigned int i = 0; i < countof(this->_alt_quic_connection_ids); ++i) {
-    this->_ctable->erase(this->_alt_quic_connection_ids[i].id, this);
-  }
-
   /* TODO: Uncmment these blocks after refactoring read / write process
     this->_udp_con        = nullptr;
     this->_packet_handler = nullptr;
     _unschedule_packet_write_ready();
 
     delete this->_handshake_handler;
     delete this->_application_map;
-    delete this->_hs_protocol;
+    delete this->_crypto;
     delete this->_loss_detector;
     delete this->_frame_dispatcher;
     delete this->_stream_manager;
     delete this->_congestion_controller;
 
-    // TODO: clear member variables like `UnixNetVConnection::free(EThread *t)`
-    this->mutex.clear();
+    super::clear();
+  */
+}
+
+// called by ET_UDP
+void
+QUICNetVConnection::remove_connection_ids()
+{
+  this->_ctable->erase(this->_original_quic_connection_id, this);
+  this->_ctable->erase(this->_quic_connection_id, this);
+  for (unsigned int i = 0; i < countof(this->_alt_quic_connection_ids); ++i) {
+    this->_ctable->erase(this->_alt_quic_connection_ids[i].id, this);
+  }
+}
 
+// called by ET_UDP
+void
+QUICNetVConnection::destroy(EThread *t)
```

Comment:
I still don't remember the difference between `free()` and `destroy()`. It's always confusing. Should we call `Allocator.free(this)` in `destroy()` but not `free()`?

Reply:
Yes we should call `Allocator.free(this)` in destroy(), since `NetHandler::free_netvc` will callack to `QVC::free` function. But we can't free memory there, so I involve the `destroy` function to free the real mem.

Reply:
Will call `NetHandler::free_netvc` in closed state. Since the vc is still in used sometimes, we need to check check the refcount to determine whether the vc could be freed in ET_UDP.

Reply:
OK... I just wanted to make sure that.